upf/conditional_effects_remover.py

```diff
@@ -0,0 +1,92 @@
+# Copyright 2021 AIPlan4EU project
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+"""This module defines the conditional effects remover class."""
+
+from collections import OrderedDict
+from upf.problem import Problem
+from upf.action import Action
+from upf.effect import Effect
+from typing import List, Dict, Tuple
+
+
+class ConditionalEffectsRemover():
+    def __init__(self, problem: Problem):
+        self._problem = problem
+        self._action_mapping: Dict[str, str] = {}
+        self._env = problem.env
+
+    def get_unconditional_problem(self) -> Problem:
+        #cycle over all the actions
+        #Note that a different environment might be needed when multy-threading
+        action_stack: List[Tuple[str, Action, int]] = []
+        new_problem = self._create_problem_copy(action_stack)
+
+        while len(action_stack) > 0:
+            original_action_name, action, number = action_stack.pop()
+            unchanged_effects = action.unconditional_effects()
+            cond_effects = action.conditional_effects()
+            effect_changed = cond_effects.pop()
+            unchanged_effects.extend(cond_effects)
+            #TODO parameters coping still missing!!
+            #TODO shallow-copy creation in it's own dedicated function!!
+            new_action_t = self._create_action_copy(action, original_action_name, number, unchanged_effects)
+            number = number + 1
+            new_action_f = self._create_action_copy(action, original_action_name, number, unchanged_effects)
+            number = number + 1
+            new_action_t.add_precondition(effect_changed.condition())
+            new_effect = Effect(effect_changed.fluent(), effect_changed.value(),
+                    self._env.expression_manager.TRUE(), effect_changed.kind())
+            new_action_t.add_effect_class(new_effect)
+            new_action_f.add_precondition(self._env.expression_manager.Not(effect_changed.condition()))
+            if new_action_t.has_conditional_effects():
+                action_stack.append((original_action_name, new_action_t, number))
```

Comment:
Is it missing the `+1` here?
```suggestion
                action_stack.append((original_action_name, new_action_t, number+1))
```

Reply:
No it should be right, the number is incremented right after usage (after the `
_create_action_copy()` function. Actually the number increment could be done inside the function but then the function would have to return a tuple (int, Action) so I preferred to replicate 1 line of code instead of making it uselessly complex.

Reply:
Ok, and in line 55 why is it needed to add 2 instead of 1?